Mathlib/Analysis/Normed/Group/Ultrametric.lean

```diff
@@ -0,0 +1,88 @@
+/-
+Copyright (c) 2024 David Loeffler. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: David Loeffler
+-/
+import Mathlib.Analysis.Normed.Group.Uniform
+import Mathlib.Topology.Algebra.Nonarchimedean.Basic
+import Mathlib.Topology.MetricSpace.Ultra.Basic
+
+/-!
+# Ultrametric normed groups
+
+We give formulations of the ultrametric triangle inequality for ultrametric spaces where the metric
+is induced by a norm; and prove that such groups are nonarchimedean groups (which is *not* true by
+definition in Mathlib – the definition of `NonarchimedeanGroup` does not refer to metrics or norms).
+-/
+
+namespace IsUltrametricDist
+
+/-- Triangle inequality for ulrametric (semi) norms on groups. -/
+@[to_additive "Triangle inequality for ulrametric (semi) norms on additive groups."]
+lemma norm_mul_le_max
+    {G : Type*} [SeminormedGroup G] [IsUltrametricDist G]
+    (x y : G) : ‖x * y‖ ≤ max ‖x‖ ‖y‖ := by
+  simpa only [dist_eq_norm_div, div_inv_eq_mul, div_one, one_mul] using dist_triangle_max x 1 y⁻¹
+
+/--
+In an commutative group with an ultrametric norm, the open balls around 1 are open subgroups.
+-/
+@[to_additive "In an commutative additive group with an ultrametric norm, the open balls around 0
+are open subgroups."]
+def ball_openSubgroup (G : Type*) [SeminormedGroup G] [IsUltrametricDist G]
```

Comment:
```suggestion
def ballOpenSubgroup (G : Type*) [SeminormedGroup G] [IsUltrametricDist G]
```

Reply:
FWIW: I haven't applied this change yet, because Yakov pointed out that one can do the same for closed balls too (closed balls are open!), and I wasn't sure what case convention to use for the closed-ball version. So for now it is "ball_openSubgroup" and "closedBall_openSubgroup", and I'm inviting @eric-wieser or anyone else to tell me what the standard convention would be.